Add verbose option

momo-the-monster opened the pull request:
Thanks for all your work on this project! I'm using it for Colab notebook experiments and found that it printed more lines to the console than I would like, so I added this -verbose flag in for myself so I could set it to 0 when I don't need as many updates.

momo-the-monster committed:
add -verbose argument to quiet most print commands

momo-the-monster committed:
add verbose propagation to CaffeLoader

Commit:
Merge remote-tracking branch 'origin/feature/add-verbose-option'

# Conflicts:
#	CaffeLoader.py
#	neural_style.py

momo-the-monster committed:
Merge branch 'master' into feature/add-verbose-option

momo-the-monster commented:
Hi @ProGamerGov! Just fixed the merged conflicts introduced with the latest and greatest from the master branch so I'm bumping this up for visibility. The -verbose flag is super helpful when running in Colab notebooks, especially as I have this in front of a group of designers at our company right now. Let me know if you have a thought on another way you'd like to see this implemented. Thanks!

Commit:
add verbose parameter to loadCaffeModel

Commit:
force add simpler downloader

ProGamerGov commented:
@momo-the-monster Nice work! 

I'm not sure about using a ton of if statements to block all the print statements. It's fine to use that once in CaffeLoader to block printing the conv layers, but it looks a bit ugly to use it for every print statement. 

I guess one could use: `sys.stdout = open(os.devnull, 'w')` to completely block printing until some point, but reenabling printing could be tricky.


Also, the verbose parameter doesn't need to use integers. It can just be a boolean flag.

momo-the-monster commented:
Hm, agreed. I don't work in Python much -what do you think about using the built-in logging library (or a third-party one like [loguru](https://github.com/Delgan/loguru)) where the user could configure what level they want to print? That way I could easily supress anything error or below.

Commit:
add log_level param, support default setting

Commit:
Merge branch 'master' of https://github.com/momo-the-monster/neural-style-pt into feature/add-verbose-option

# Conflicts:
#	CaffeLoader.py
#	neural_style.py

Commit:
Merge branch 'feature/add-verbose-option' of https://github.com/momo-the-monster/neural-style-pt into feature/add-verbose-option

# Conflicts:
#	CaffeLoader.py
#	neural_style.py

momo-the-monster commented:
@ProGamerGov - had another go at this, much simpler this time around.

I just added a **-log_level** parameter and switched all the **print()** commands to **logging.info()** commands. 

The log level is INFO by default, so everything shows up. If you pass -log_level 30 or higher, then you get no outputs. Future-compatible because any new code can just log using the appropriate level, the param supports all [Python log levels](https://docs.python.org/2/library/logging.html).

It looks like you've got a couple of branches in various states so I made this off the latest master. Happy to redo it off another branch if that makes it more likely to get merged.
